### PR TITLE
CDRIVER-3909 topology description copy/destroy methods

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/CDRIVER-3909

### CODE REVIEW

File: src/libmongoc/doc/mongoc_topology_description_destroy.rst [hunk 1]
```diff
@@ -0,0 +1,22 @@
+:man_page: mongoc_topology_description_destroy
+
+mongoc_topology_description_destroy()
+===================================
```
Comment (alcaeus) [inline src/libmongoc/doc/mongoc_topology_description_destroy.rst:4]:
```suggestion
=====================================
```

File: src/libmongoc/doc/mongoc_topology_description_new_copy.rst [hunk 2]
```diff
@@ -0,0 +1,28 @@
+:man_page: mongoc_topology_description_new_copy
+
+mongoc_topology_description_new_copy()
+====================================
```
Comment (alcaeus) [inline src/libmongoc/doc/mongoc_topology_description_new_copy.rst:4]:
```suggestion
======================================
```

**Review by alcaeus — APPROVED** [review]
LGTM with suggestions applied.

**Review by kevinAlbs — APPROVED** [review]
LGTM!

**Comment by jmikola** [comment]
@tanlisu: This was just rebased on master. That should help when updating the submodule in https://github.com/mongodb/mongo-php-driver/pull/1244.

**Comment by jmikola** [comment]
Note: the ASAN failures here appear to be legitimate issues so I'm going to spend some more time on investigating and fixing this. For the time being I'm going to rebase on an earlier libmongoc commit before https://github.com/mongodb/mongo-c-driver/commit/fdc2de4ff0e472a03c6c8a22dccaa1916ef3fd08 to unblock @tanlisu in https://github.com/mongodb/mongo-c-driver/pull/785.

**Pull request merged**